app/factory/Layer.js

```diff
@@ -88,22 +88,45 @@ Ext.define('CpsiMapview.factory.Layer', {
      * @return {ol.layer.Tile}    WMS layer
      */
     createWms: function(layerConf) {
-        return new ol.layer.Tile({
-            name: layerConf.text,
-            source: new ol.source.TileWMS({
-                url: layerConf.url,
-                params: {
-                    'LAYERS': layerConf.serverOptions.layers,
-                    'TILED': true,
-                    'TRANSPARENT': true
-                },
-                crossOrigin: 'anonymous'
-            }),
-            visible: layerConf.openLayers.visibility,
-            minResolution: layerConf.openLayers.minResolution,
-            maxResolution: layerConf.openLayers.maxResolution,
-            opacity: layerConf.openLayers.opacity
-        });
+        var layer;
+        var singleTile = layerConf.openLayers.singleTile;
+
+        if (singleTile) {
+            layer = new ol.layer.Image({
+                name: layerConf.text,
+                source: new ol.source.ImageWMS({
+                    url: layerConf.url,
+                    params: {
+                        'LAYERS': layerConf.serverOptions.layers,
+                        'TRANSPARENT': true
+                    },
+                    ratio: 1
+                }),
+                visible: layerConf.openLayers.visibility,
+                minResolution: layerConf.openLayers.minResolution,
+                maxResolution: layerConf.openLayers.maxResolution,
+                opacity: layerConf.openLayers.opacity
+            });
+        } else {
+            layer = new ol.layer.Tile({
+                name: layerConf.text,
+                source: new ol.source.TileWMS({
+                    url: layerConf.url,
+                    params: {
+                        'LAYERS': layerConf.serverOptions.layers,
+                        'TILED': true,
+                        'TRANSPARENT': true
+                    },
+                    crossOrigin: 'anonymous'
+                }),
+                visible: layerConf.openLayers.visibility,
+                minResolution: layerConf.openLayers.minResolution,
+                maxResolution: layerConf.openLayers.maxResolution,
+                opacity: layerConf.openLayers.opacity
+            });
+        }
```

Comment:
This has some duplicated code when basically everything is the same, except for the constructors that are used and one param (`'TILED': true`) in a config object. (I assume that `crossOrigin` should be present in both `if`-branches) 

This is perfectly fine as it is, in case you want to change it to have less duplicated parts, I agree. I'll let you decide which way you go, because in the current way the readability probably is a bit better.

Reply:
I agree and I plan to harmonize some parts of the property applying at all, since some are duplicated in other functions as well. For the moment it gives a better overview which config properties are already processed.